ludwig/data/dataframe/dask.py

```diff
@@ -129,7 +133,25 @@ def to_ray_dataset(self, df):
         return from_dask(df)
 
     def from_ray_dataset(self, dataset) -> dd.DataFrame:
-        return dataset.to_dask()
+        """Custom Ray to Dask conversion implementation to pass in meta during dd.DataFrame creation."""
+
+        @dask.delayed
+        def block_to_df(block: Block):
+            block = BlockAccessor.for_block(block)
+            if isinstance(block, (ray.ObjectRef, ClientObjectRef)):
+                raise ValueError(
+                    "Dataset.to_dask() must be used with Dask-on-Ray, please "
+                    "set the Dask scheduler to ray_dask_get (located in "
+                    "ray.util.dask)."
+                )
+            return block.to_pandas()
+
+        # Use a sample of the ray dataset to generate meta to infer types even if there are NaNs
+        meta = dataset.limit(META_SAMPLE_SIZE).to_pandas()
+        # HACK: Dask cannot handle TensorDtype yet, so we ensure those are of dtype object
```

Comment:
Is there a TODO/issue we can point to track when Dask will be able to handle TensorDtype?

Reply:
@justinxzhao I just checked Dask's GitHub and I don't see any existing reported issues related to handling TensorDType. This check is only relevant for Ray 2.0 and the latest nightly releases, so essentially Dask on Ray. Would it make sense to flag it on both Ray and Dask's ends? Or just Dask?

Reply:
Ideally, I'm hoping that the Ray team updates their `to_dask()` function in the future to handle this neatly.

Reply:
FYI, Dask does not have support for Pandas extension types in general, it looks like. https://github.com/dask/dask/issues/5720